django/db/models/options.py

```diff
@@ -200,6 +202,11 @@ def contribute_to_class(self, cls, name):
 
             self.unique_together = normalize_together(self.unique_together)
             self.index_together = normalize_together(self.index_together)
+            if self.index_together:
+                warnings.warn(
+                    "'index_together' is deprecated in favor of 'indexes'",
```

Comment:
The main question is what to do with these tests? We should analyze them one by one and prepare alternative versions only with `unique_together` (if necessary). I'm afraid that we cannot simply remove them when deprecation ends because we will end with many not covered scenarios. For example:
```python
diff --git a/tests/migrations/test_autodetector.py b/tests/migrations/test_autodetector.py
index 547e0b32c5..4c19a34d7f 100644
--- a/tests/migrations/test_autodetector.py
+++ b/tests/migrations/test_autodetector.py
@@ -2441,10 +2441,10 @@ class AutodetectorTests(TestCase):
         self.assertNumberMigrations(changes, "testapp", 1)
         self.assertOperationTypes(changes, "testapp", 0, ["AlterField"])
 
+    # RemovedInDjango51Warning: When deprecation ends rename to
+    # test_empty_unique_together().
+    @ignore_warnings(category=RemovedInDjango51Warning)
     def test_empty_foo_together(self):
-        """
-        #23452 - Empty unique/index_together shouldn't generate a migration.
-        """
         # Explicitly testing for not specified, since this is the case after
         # a CreateModel operation w/o any definition on the original model
         model_state_not_specified = ModelState(
@@ -2457,7 +2457,7 @@ class AutodetectorTests(TestCase):
             "model",
             [("id", models.AutoField(primary_key=True))],
             {
-                "index_together": None,
+                "index_together": None,  # RemovedInDjango51Warning
                 "unique_together": None,
             },
         )
@@ -2468,7 +2468,7 @@ class AutodetectorTests(TestCase):
             "model",
             [("id", models.AutoField(primary_key=True))],
             {
-                "index_together": set(),
+                "index_together": set(),  # RemovedInDjango51Warning
                 "unique_together": set(),
             },
         )
```

Reply:
> Maybe we can have a different PR (or even ticket) to split the tests and make the `index_together` tests redundant with `unique_together` and easily delete-able.

Yes, please :bow: , we could move `index_together` tests to a separate class, e.g. `AutodetectorIndexTogetherTests`. A different PR with `Refs #27236 -- ...` would be fine.

Reply:
See https://github.com/django/django/pull/15755